core/Property/Settings/Storage.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Property\Settings;
+
+use Piwik\Db;
+use Piwik\Common;
+use Piwik\Settings\Setting;
+
+/**
+ * Storage for site settings
+ */
+class Storage extends \Piwik\Settings\Storage
+{
+    private $idSite = null;
+
+    /**
+     * @var Db
+     */
+    private $db = null;
+
+    private $toBeDeleted = array();
+
+    public function __construct(Db\AdapterInterface $db, $idSite)
+    {
+        $this->db     = $db;
+        $this->idSite = $idSite;
+    }
+
+    protected function deleteSettingsFromStorage()
+    {
+        $table = $this->getTableName();
+        $sql   = "DELETE FROM $table WHERE `idsite` = ?";
+        $bind  = array($this->idSite);
+
+        $this->db->query($sql, $bind);
```

Comment:
Didn't create a separate class for the SQL stuff as we will refactor it soon anyway.